File: frontend/src/app/shared/responses/multiple-client.response.ts
export class MultipleClientResponse {
  clientDetails: ClientDetails[];

  constructor(clientDetails: ClientDetails[]) {
    this.clientDetails = clientDetails;
  }

}

export class ClientDetails {

  id: number;
  name: string;
  phone: string;
  email: string;

  constructor(id: number, name: string, phone: string, email: string) {
    this.id = id;
    this.name = name;
    this.phone = phone;
    this.email = email;
  }

}

File: src/main/java/com/sgorokh/TimeSaver/controllers/responses/MultipleClientResponse.java
package com.sgorokh.TimeSaver.controllers.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
public class MultipleClientResponse {

    private List<ClientDetails> clientDetails;

    @Builder
    @Data
    public static class ClientDetails {
        private Long id;
        private String name;
        private String phone;
        private String email;
    }

}
